app/scripts/controllers/transactions.js

```diff
@@ -201,6 +210,7 @@ module.exports = class TransactionController extends EventEmitter {
       nonceLock.releaseLock()
     } catch (err) {
       this.setTxStatusFailed(txId, {
```

Comment:
we can still just pass the error but in some instances we will still need to write are own errors